src/ServicePulse.Host/vue/src/components/OrderBy.vue

```diff
@@ -1,17 +1,32 @@
+<script lang="ts">
+export function getSortFunction<T>(selector: SortOptions<T>["selector"], dir: SortDirection) {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  if (!selector) return (firstElement: T, secondElement: T) => 0;
+  return (firstElement: T, secondElement: T) => {
+    if (dir === SortDirection.Ascending) {
+      return selector(firstElement) < selector(secondElement) ? -1 : 1;
+    } else {
+      return selector(firstElement) < selector(secondElement) ? 1 : -1;
+    }
+  };
+}
+</script>
```

Comment:
Why is this in it's own script block?

Reply:
because you can't export from a script setup block

Reply:
Are you sure? We have done that elsewhere I thought

Reply:
it appears that you can with types (since they compile out anyway) but not actual code

`ERROR(ESLint)  `<script setup>` cannot contain ES module exports. (vue/no-export-in-script-setup)`